purchase_order_no_owner/models/purchase_order.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    no_owner = fields.Boolean()
+
+    @api.constrains("no_owner", "owner_id")
+    def _check_owner_id(self):
```

Comment:
@rinaldifirdaus I think we should actually do this check when confirming PO instead.  Making it a constraint may introduce an inconvenience of having to be sure about the arrangement at the very beginning.  Users may just select No Owner when they are not sure when they create a draft PO, and then later forget to update the settings even when it later turns out an owner should be assigned.